parsers/json/json.cc

```diff
@@ -0,0 +1,108 @@
+#include "json.h"
+
+namespace
+{
+  using namespace trieste;
+  using namespace trieste::json;
+
+  std::size_t invalid_tokens(
+    Node n, std::initializer_list<Token> tokens, const std::string& message)
+  {
+    std::size_t changes = 0;
+    for (auto child : *n)
+    {
+      if (child->in(tokens))
+      {
+        n->replace(child, err(child, message));
+        changes += 1;
+      }
+      else
+      {
+        changes += invalid_tokens(child, tokens, message);
+      }
+    }
+
+    return changes;
+  }
+}
+
+namespace trieste::json
+{
+
+  const auto ValueTokens = T(Object, Array, String, Number, True, False, Null);
```

Comment:
This only matches one token, so should it be:
```suggestion
  const auto ValueToken = T(Object, Array, String, Number, True, False, Null);
```

Reply:
Fixed.